packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,202 @@ def apply_default_cascade(args):
         args["marginal_x"] = None
 
 
+def _name_heuristic(argument, field_name, used_col_names):
+    if isinstance(argument, int):
+        argument = str(argument)
+    if field_name not in used_col_names:
+        return field_name
+    elif field_name + argument not in used_col_names:
+        return field_name + "_" + argument
+    else:
+        raise NameError(
+            "A name conflict was encountered for argument %s."
+            "Columns with names %s, %s and %s are already used"
+            % (field_name, argument, field_name, field_name + "_" + argument)
+        )
+
+
+def _initialize_argument_col_names(args, attrables, array_attrables):
+    """
+    This function builds a list of columns of the data_frame argument used
+    as arguments, either as str/int arguments or given as columns
+    (pandas series type).
+    """
+    df = args["data_frame"]
+    used_col_names = set()
+    for field in args:
+        if field not in attrables:
+            continue
+        names = args[field] if field in array_attrables else [args[field]]
+        if names is None:
+            continue
+        for arg in names:
+            if arg is None:
+                continue
+            if isinstance(arg, str):
```

Comment:
what if `arg` isn't in `df` ?

Reply:
it will be caught later on, L882 
```
 if argument not in args["data_frame"].columns:
                    raise ValueError(
                        "Value of '%s' is not the name of a column in 'data_frame'. "
                        "Expected one of %s but received: %s"
                        % (field, str(list(args["data_frame"].columns)), argument)
                    )
```
is it ok for you?

Reply:
yes, I was just curious about why it ends up in the reserved names

Reply:
we could also raise the error in the function building the list of reserved names, which is always called